src/pybind/mgr/dashboard/services/access_control.py

```diff
@@ -181,13 +181,26 @@ def __init__(self, username, password, name=None, email=None, roles=None,
             self.refreshLastUpdate()
         else:
             self.lastUpdate = lastUpdate
+        self.needChangePassword = needChangePassword
 
     def refreshLastUpdate(self):
         self.lastUpdate = int(time.time())
 
     def set_password(self, password):
+        if bcrypt.checkpw(password.encode('utf8'), self.password.encode('utf8')):
+            raise DashboardException(msg='New password is the same as the current one.',
+                                 code='same-password',
+                                 component='Update User')
         self.password = password_hash(password)
         self.refreshLastUpdate()
+        if self.needChangePassword:
+            self.update_needChangePassword()
+            
+    def update_needChangePassword(self):
+        self.needChangePassword = not self.needChangePassword
```

Comment:
IMHO this is rather tricky. This basically flip-flops between `True` and `False`, so you cannot set the desired state of the flag. Why not explicitly setting it to a specific value?

Optionally, as this is basically a `setter`-like method, why not using `@property`:
```python
@property
def change_password(self):
    return self._change_password
 
@change_password.setter
def change_password(self, value):
  self._change_password = value
  self.refreshLastUpdate()
```